File: cases/taint/cpp/exception-catch-positive/exception_catch_positive.cpp
struct FlowException {
    int value;
};

int dfb_source() { // DFB-SOURCE: exception-catch-input
    return 1;
}

void dfb_sink(int value) {} // DFB-SINK: exception-catch-sink

void run() {
    try {
        FlowException flow;
        flow.value = dfb_source();
        throw flow; // DFB-WITNESS: exception-catch-throw
    } catch (FlowException &caught) {
        dfb_sink(caught.value);
    }
}

File: cases/taint/csharp/exception-catch-positive/ExceptionCatchPositive.cs
using System;

namespace DataFlowBench;

static class ExceptionCatchPositive
{
    sealed class FlowException : Exception
    {
        public int Value;
    }

    static int dfb_source() // DFB-SOURCE: exception-catch-input
    {
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: exception-catch-sink

    static void Run()
    {
        try
        {
            FlowException flow = new FlowException();
            flow.Value = dfb_source();
            throw flow; // DFB-WITNESS: exception-catch-throw
        }
        catch (FlowException caught)
        {
            dfb_sink(caught.Value);
        }
    }
}
